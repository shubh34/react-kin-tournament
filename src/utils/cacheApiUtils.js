export const toBeCachedApiResponse = ['GET_TOURNAMENT_LEADERS_REQUEST', 'GET_TOURNAMENT_PRIZES_REQUEST'];

export const cacheApiResponse = (response) => {
	const { meta: { cacheKey = '' }, payload } = response;
	if (toBeCachedApiResponse.includes(cacheKey) && !sessionStorage.getItem(cacheKey)) {
		const responseToBeCached = {
			createdTimeStamp: Date.now(),
			response: payload,
		};
		sessionStorage.setItem(cacheKey, JSON.stringify(responseToBeCached));
	}
};
