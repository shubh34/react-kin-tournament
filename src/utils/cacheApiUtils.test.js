import { cacheApiResponse } from './cacheApiUtils';

Date.now = jest.fn(() => 1566735528409);

describe('cacheApiUtils', () => {
	it('should not cache the Api response when its not in toBeCachedApiResponse list', () => {
		const response = {
			meta: { cacheKey: 'NOT_LISTED_API_RESPONSE' },
			payload: { test: 'new Response' },
		};
		cacheApiResponse(response);
		expect(sessionStorage).toMatchSnapshot();
	});

	it('should cache the Api response', () => {
		const response = {
			meta: { cacheKey: 'GET_TOURNAMENT_LEADERS_REQUEST' },
			payload: { test: 'test' },
		};
		cacheApiResponse(response);
		expect(sessionStorage).toMatchSnapshot();
	});
	it('should not cache the Api response when already cached', () => {
		const response = {
			meta: { cacheKey: 'GET_TOURNAMENT_LEADERS_REQUEST' },
			payload: { test: 'new Response' },
		};
		cacheApiResponse(response);
		expect(sessionStorage).toMatchSnapshot();
	});
});
