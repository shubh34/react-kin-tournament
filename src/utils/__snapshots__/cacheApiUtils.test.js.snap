// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cacheApiUtils should cache the Api response 1`] = `
Storage {
  "GET_TOURNAMENT_LEADERS_REQUEST": "{\\"createdTimeStamp\\":1566735528409,\\"response\\":{\\"test\\":\\"test\\"}}",
}
`;

exports[`cacheApiUtils should not cache the Api response when already cached 1`] = `
Storage {
  "GET_TOURNAMENT_LEADERS_REQUEST": "{\\"createdTimeStamp\\":1566735528409,\\"response\\":{\\"test\\":\\"test\\"}}",
}
`;

exports[`cacheApiUtils should not cache the Api response when its not in toBeCachedApiResponse list 1`] = `Storage {}`;
